src/guest.S

```diff
@@ -43,6 +43,10 @@ _run_guest:
     csrrw t1, hstatus, t1
     sd    t1, ({host_hstatus})(a0)
 
+    ld    t1, ({guest_hcounteren})(a0)
+    csrrw t1, hcounteren, t1
```

Comment:
@abrestic-rivos I am not sure what exactly do you mean "It only applies when V=1". It's a hypervisor CSR not guest CSR.
Thus, hypervisor can update it whenever it wants. It is more similar to hstatus/hedeleg/hideleg

Reply:
Sorry, I should've been more precise here: `hcounteren` has no effect on the current execution environment unless V=1 since it is simply enabling access to the specified counters in VS/VU. So there is no need to swap it from assembly.